### ISSUE TITLE
Helm Lint doesn't catch number value for ConfigMap

### ISSUE THREAD

**Issue by busches** [helm/helm#11620]
Output of `helm version`: `version.BuildInfo{Version:"v3.10.2", GitCommit:"50f003e5ee8704ec937a756c646870227d7c8b58", GitTreeState:"clean", GoVersion:"go1.19.3"}`

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:28:30Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Rancher

Kubernetes requires config map properties to be strings, but when you have an unquoted number, no error is returned by `helm lint`

Example:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: myconfig
data:
  PROPERTY_ONE: 100
  PROPERTY_TWO: {{ .Values.propertyTwo }}
```

Values Yaml would have:
```
propertyTwo: 101
```

Full example: [helm-example.zip](https://github.com/helm/helm/files/10195843/helm-example.zip)

Error returned when applying the ConfigMap:
```
cannot patch "label-batch-manifest" with kind ConfigMap: "" is invalid: patch: Invalid value: "{\"apiVersion\":\"v1\",\"data\":{\"PROPERTY_ONE\":100,\"PROPERTY_TWO\":101,\"kind\":\"ConfigMap\",\....": json: cannot unmarshal number into Go struct field 
ConfigMap.data of type string
```

**Comment by gjenkins8** [comment]
I think the `helm lint` docs could be improved as to what it actually does. `helm lint` today does things like [e.g.](https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/lint/rules/chartfile.go#L57) verifying a `Chart.yaml` has an API version. 

I don't believe it verifies templated manifests against Kubernetes API types. There are also subtles here. Like requiring an API server (traditionally, a 'lint' tool wouldn't require a live environment). To verifying custom resource types (which may not have an installed definition).

(personally, I have used something like `helm template | kubectl apply --dry-run` to achieve similar)

**Comment by busches** [comment]
@gjenkins8 Agreed, except, `kubectl apply --dry-run` also doesn't find this error. 🤷

**Comment by joejulian** [comment]
What about `kubectl apply --dry-run=server`?

**Comment by busches** [comment]
@joejulian That did indeed catch it, that will improve our quality of life.

**Comment by joejulian** [comment]
Happy I could help.

**Comment by thw0rted** [comment]
I know this issue was closed out a while back, but it just bit us.  I felt pretty good when I set up our CI to run `helm template` against all our charts, but still had to dig through a bunch of output to figure out why we were getting

> Error: ConfigMap in version "v1" cannot be handled as a ConfigMap: json: cannot unmarshal number into Go struct field ConfigMap.data of type string

even though the CI job that runs lint/template passed.

I don't want to have to run my CI job with a connection to the production server, is there any other way to set up a test during CI that will fail if a `data` value isn't a string?
